bftengine/src/bftengine/SimpleClientImp.cpp

```diff
@@ -589,7 +592,7 @@ void SimpleClientImp::sendRequestToAllOrToPrimary(bool sendToAll, char* data, ui
   } else {
     LOG_DEBUG(logger_, "Send request to primary replica" << KVLOG(clientId_, firstReqSeqNum));
     pm_->Delay<concord::performance::SlowdownPhase::BftClientBeforeSendPrimary>();
-    communication_->send(knownPrimaryReplica_, std::move(msg));
+    communication_->send(knownPrimaryReplica_, std::move(msg), knownPrimaryReplica_);
```

Comment:
Own principal id should be passed in the last parameter.

Reply:
done.